From 72f8f5650557e0555113ce02ad67e43f8b76eed0 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 14 May 2019 16:02:19 +0800
Subject: [PATCH] logging: fix /dev/log label

When using systemd, /dev/log is a symlink to
/run/systemd/journal/dev-log

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/logging.fc | 1 +
 policy/modules/system/logging.if | 4 ++++
 policy/modules/system/logging.te | 1 +
 3 files changed, 6 insertions(+)

diff --git a/policy/modules/system/logging.fc b/policy/modules/system/logging.fc
index 5bec7e9..39ede06 100644
--- a/policy/modules/system/logging.fc
+++ b/policy/modules/system/logging.fc
@@ -1,4 +1,5 @@
 /dev/log		-s	gen_context(system_u:object_r:devlog_t,mls_systemhigh)
+/dev/log		-l	gen_context(system_u:object_r:devlog_t,mls_systemhigh)
 
 /etc/rsyslog\.conf					--	gen_context(system_u:object_r:syslog_conf_t,s0)
 /etc/syslog\.conf					--	gen_context(system_u:object_r:syslog_conf_t,s0)
diff --git a/policy/modules/system/logging.if b/policy/modules/system/logging.if
index dcb1bd5..2d70e76 100644
--- a/policy/modules/system/logging.if
+++ b/policy/modules/system/logging.if
@@ -623,6 +623,7 @@ interface(`logging_send_syslog_msg',`
 	')
 
 	allow $1 devlog_t:sock_file write_sock_file_perms;
+	allow $1 devlog_t:lnk_file rw_lnk_file_perms;
 
 	# systemd journal socket is in /run/systemd/journal/dev-log
 	init_search_run($1)
@@ -662,6 +663,7 @@ interface(`logging_relabelto_devlog_sock_files',`
 	')
 
 	allow $1 devlog_t:sock_file relabelto_sock_file_perms;
+	allow $1 devlog_t:lnk_file relabelto_lnk_file_perms;
 ')
 
 ########################################
@@ -681,6 +683,8 @@ interface(`logging_create_devlog',`
 
 	allow $1 devlog_t:sock_file manage_sock_file_perms;
 	dev_filetrans($1, devlog_t, sock_file)
+	allow $1 devlog_t:lnk_file manage_lnk_file_perms;
+	dev_filetrans($1, devlog_t, lnk_file)
 	init_pid_filetrans($1, devlog_t, sock_file, "syslog")
 ')
 
diff --git a/policy/modules/system/logging.te b/policy/modules/system/logging.te
index e6ef7ce..71fc040 100644
--- a/policy/modules/system/logging.te
+++ b/policy/modules/system/logging.te
@@ -410,6 +410,7 @@ allow syslogd_t syslog_conf_t:dir list_dir_perms;
 
 # Create and bind to /dev/log or /var/run/log.
 allow syslogd_t devlog_t:sock_file manage_sock_file_perms;
+allow syslogd_t devlog_t:lnk_file manage_lnk_file_perms;
 files_pid_filetrans(syslogd_t, devlog_t, sock_file)
 init_pid_filetrans(syslogd_t, devlog_t, sock_file, "dev-log")
 
-- 
2.7.4

